### PR TITLE
Remove metadata in dtype_decode function. It fixes gh-1152, because h…

### LINKED ISSUE #1152: Error converting hdf5 file to zarr
### Zarr version

2.13.2

### Numcodecs version

0.10.2

### Python Version

3.10.4

### Operating System

Mac

### Installation

pip

### Description

 TypeError: Cannot compare structured or void to non-void arrays when converting h5 file to zarr

### Steps to reproduce

```
import urllib

import h5py
import zarr


testfile = urllib.URLopener()
testfile.retrieve(
    "https://raw.githubusercontent.com/chanzuckerberg/cellxgene/main/example-dataset/pbmc3k.h5ad",
    "pbmc3k.h5ad",
)
with h5py.File("pbmc3k.h5ad", "r") as h5_file:
    zarr.copy_all(h5_file, zarr.open("pbmc3k.zarr", "w"))
```

### Additional output

_No response_

### PR DESCRIPTION
**Pull request by ninousf**
h5py puts metadata in dtypes which are not correctly managed in numpy

revival of #1349
fixes #1152

### CODE REVIEW

**Comment by ninousf** [comment]
No answer. Did i do something wrong ?

**Comment by joshmoore** [comment]
Hi @ninousf. Apologies for the slow response. Speaking for myself, I was on the road for 6 weeks and am trying now to get caught up.

Note: I've added "fixes #1152" to the description.

**Comment by joshmoore** [comment]
Thanks again, @ninousf. I've pushed a changelog entry for your review.

Finally opinions welcome, otherwise looking forward to getting this in 2.15.x.

File: zarr/meta.py [hunk 1]
```diff
@@ -187,6 +187,12 @@ def encode_dtype(cls, d: np.dtype):
     @classmethod
     def _decode_dtype_descr(cls, d) -> List[Any]:
         # need to convert list of lists to list of tuples
+
+        # remove metadata in descr
+        if isinstance(d, (list, tuple)) and len(d) == 2 and isinstance(
+                d[1], dict):
+            return d[0]
```
Comment (joshmoore) [inline zarr/meta.py:194]: Is it possible that this needs the recursive handling of the next section, @ninousf?

**Comment by jhamman** [comment]
This has unfortunately gone stale. Seems like it was quite close. Will it be revived or should we close this?

**Comment by joshmoore** [comment]
Might require a new champion then ;)

**Pull request closed without merging**